dali/pipeline/pipeline.cc

```diff
@@ -476,23 +478,27 @@ void Pipeline::Build(vector<std::pair<string, string>> output_names) {
           name + "' only exists on gpu.");
 
       if (!it->second.has_contiguous) {
-        // Add a make contiguous op to produce this output
-        OpSpec spec =
-          OpSpec("MakeContiguous")
-          .AddArg("device", "mixed")
-          .AddInput(name, "cpu")
-          .AddOutput("contiguous_" + name, "cpu");
-        PrepareOpSpec(&spec, GetNextInternalLogicalId());
-
-        graph_.AddOp(spec, "__MakeContiguous_" + name);
+        // Don't add MakeContiguous twice for the same graph output in case they are duplicated
+        if (unique_outputs.find(name + device) == unique_outputs.end()) {
+          // Add a make contiguous op to produce this output
+          OpSpec spec =
+            OpSpec("MakeContiguous")
+            .AddArg("device", "mixed")
+            .AddInput(name, "cpu")
+            .AddOutput("contiguous_" + name, "cpu");
+          PrepareOpSpec(&spec, GetNextInternalLogicalId());
+
+          graph_.AddOp(spec, "__MakeContiguous_" + name);
```

Comment:
```suggestion
          graph_.AddOp(spec, "__MakeContiguous_" + name);
          it->second.has_contiguous = true;
```
This should solve the issue without the need for an extra `unordererd_set` - and I think it's missing anyway.

Reply:
Done